src/dropbox.js

```diff
@@ -0,0 +1,85 @@
+import dropbox from 'dropbox';
+
+const dbx = {
+
+    // Must be called in componentWillMount in App.js
+    initialize: function (callback) {
+        var config = {
+            appKey: "l3bfhq15xjjtxqp",
+            clientId: "jhbdlvkjabsdkljvna",
+            clientSecret: "t523msf8d0pr610",
+            redirectUri: "http://localhost:3000/"
+        };
+        let accessToken = this.getAccessTokenFromUrl();
+        if (!accessToken) {
+            window.location.href = "http://dropbox.com/oauth2/authorize?client_id=" + config.appKey + "&response_type=token&redirect_uri=" + window.location.href;
+        } else  {
+            config.accessToken = accessToken;
+            this.app = new dropbox.Dropbox(config);
+        }
+    },
+
+    getAccessTokenFromCode: function (redirectUri, code) {
+        this.app.getAccessTokenFromCode(redirectUri, code)
+            .then(function(token) {
+                console.log(token);
+            })
+            .catch(function(error) {
+                console.log(error);
+            });
+    },
+
+     // Parses the url and gets the access token if it is in the urls hash
+    getAccessTokenFromUrl: function () {
+        return parseQueryString(window.location.hash).access_token;
+    },
+
+    isAuthenticated: function () {
+        return !!this.getAccessTokenFromUrl();
+    },
+
+    openFile: function (fileName) {
+        window.location.href = "https://www.dropbox.com/home?preview=" + fileName;
+    }
+
+}
+
+function parseQueryString(str) {
+    var ret = Object.create(null);
```

Comment:
https://stackoverflow.com/questions/42862253/how-to-parse-query-string-in-react-router-v4

Reply:
ty, done